### PR TITLE
[ANCHOR-378] Status callback retry implementation.

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

The event delivery should retry in one of the 3 ways:

1. If connection error (IOException) happens, we will retry 3 times with `networkBackoffTimer`. After that we will skip the event and proceed to next sub-sequent events.

1. If the business server returns HTTP status code other than 200s or 300s, we will retry 3 times with `httpErrorBackoffTimer`. After 3 retries, we will skip the event and proceed to next sub-sequent events.

1. If the event processor encounters an un-expected error caused by the un-seen bugs, the event will be delivered to a no-op DLQ class which has Log.error implementation. 

### Why

Implement reliable retry.

### Known limitations

Needs unit tests.

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java [hunk 1]
```diff
@@ -74,43 +77,45 @@ public void start() {
...
-            continue;
-          }
+              new CallbackApiEventHandler(callbackApiConfig),
+              eventService));
+    }
+    // Create a processor of the client status callback handler for each client defined in the
+    // clientsConfig
+    if (eventProcessorConfig.getClientStatusCallback().isEnabled()) {
+      for (ClientsConfig.ClientConfig clientConfig : clientsConfig.getClients()) {
+        if (clientConfig.getCallbackUrl().isEmpty()) {
```
Comment (JiahuiWho) [inline platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java:87]: Nit:  if( **_!_** clientConfig.getCallbackUrl().isEmpty()) {...}

**Review by JiahuiWho — APPROVED** [review]

**Review by JakeUrban — COMMENTED** [review]
I know we've discussed this a few times, but I'm still not sure it makes sense to have different retry behavior depending on if we experience a connection error or http error. AWS SNS via HTTP doesn't seem to distinguish between these cases.

Unrelatedly, I noticed that `BaseApiClient`'s `client` object's timeouts seem way too long. I would have expected something around 15 seconds, rather than 10 minutes.

**Comment by lijamie98** [comment]
@philipliu Please feel free to take care of Jake's comments and submit commits during my absence.

**Comment by JiahuiWho** [comment]
Just curious that how important it is if we missed delivering an event? With current implementation, we could miss delivering an event with http error, but why not with connection error (since we keep retrying until succeed)? If clients service is down, they should be able to pull all events  during the downtime by themselves?

**Comment by JakeUrban** [comment]
I think thats the core of the argument I'm trying to make; it is the client & business' responsibility to ensure events are received and processed correctly, and if they don't, assuming it is not the fault of the anchor platform itself due to an unexpected error, the client or business should have a system in place to ensure it eventually gets that update.

When we built the original version of the SDP, we listened for Circle's SNS notifications to get updates on payments we initiated via their API. In case we missed an event or didn't receive it, we also had a reoccurring check for transactions that hadn't been updated within a timeframe we felt was reasonable. If we found transactions that hadn't been updated in the configured amount of time, we'd make requests to Circle's API to get the latest status.

Now, clients and businesses wouldn't have to implement that logic if the Anchor Platform retried indefinitely for all events, but right now it is only making that guaranteed for a subset of events. I don't think retrying indefinitely is a viable approach in the long-term, so I think we should use the same logic we have for http errors for connection errors.

**Comment by JiahuiWho** [comment]
Thanks for clarification. I will take care of this

**Review by JakeUrban — APPROVED** [review]
This looks good @JiahuiWho ! Can you update the PR description to accurately describe the functionality? After that we should be good to merge 🚀

**Pull request merged**